apps/bot/src/services/streamstats/index.ts

```diff
@@ -0,0 +1,180 @@
+import { prisma, prismaQueue, storage } from "services/db";
+import { TwitchApi } from "services/twitch-api";
+import type { TTwitchApiChatter, TTwitchIrcContext } from "services/types";
+import { logger } from "utils/logger";
+
+export class StreamStatsGatherer {
+  public static instances = new Map<string, StreamStatsGatherer>();
+
+  private api: TwitchApi;
+
+  private activeStream: string | undefined = undefined;
+  private activeStreamId = -1;
+
+  constructor(private channel: string) {
+    this.api = TwitchApi.getInstance(channel);
+  }
+
+  public static getInstance(ch: string): StreamStatsGatherer {
+    const instance = StreamStatsGatherer.instances.get(ch);
+
+    if (instance) {
+      return instance;
+    }
+
+    const newInstance = new StreamStatsGatherer(ch);
+
+    StreamStatsGatherer.instances.set(ch, newInstance);
+    return newInstance;
+  }
+
+  private statsInterval: Timer | undefined = undefined;
+  private standbyInterval: Timer | undefined = undefined;
+
+  private async startGathering(startedAt: string, streamerId: string, interval = 30_000) {
+    logger.info(`${this.channel} is now live, starting gathering stats`);
+    storage.set(`${this.channel}_is_live`, true);
+
+    storage.set(this.messagesKey, 0);
+    storage.set(this.newChattersKey, 0);
+    this.activeStream = `${startedAt}@${this.channel}`;
+
+    const res = await prisma.streams.upsert({
+      where: { unique_id: this.activeStream },
+      create: {
+        started_at: startedAt,
+        channel: this.channel,
+        unique_id: this.activeStream,
+        streamer_id: streamerId,
+      },
+      update: { started_at: startedAt },
+    });
+
+    this.activeStreamId = res.id;
+
+    this.statsInterval = setInterval(() => {
+      this.handleGathering();
+    }, interval);
+  }
+
+  public async startStandby() {
+    logger.info(`${this.channel} is now offline, standby mode`);
+    storage.set(`${this.channel}_is_live`, false);
+    clearInterval(this.statsInterval);
+
+    this.standbyInterval = setInterval(async () => {
+      await this.handleStandby();
+    }, 60_000);
```

Comment:
This `60_000` is a magic number here, maybe move this to a private field so you can keep record of it, and update easier later on.